libwget/hash_printf.c

```diff
@@ -64,16 +64,17 @@ void wget_hash_printf_hex(wget_digest_algorithm algorithm, char *out, size_t out
 	va_end(args);
 
 	if (plaintext) {
-		unsigned char digest[wget_hash_get_len(algorithm)];
+		size_t digestLen = wget_hash_get_len(algorithm);
+		unsigned char * digest = malloc(digestLen * sizeof(char));
```

Comment:
Why not simply `alloca()`?

Reply:
100%, happy to go through all the changes and swap them to alloca if you folks are open to potentially changing away from variable length arrays

Reply:
IMO there is no technical difference between VLA and alloca(), the latter feels just a bit clumsy. Or does alloca() check for stack boundaries ?

@mitchcapper Is there a technical reason to use MSVC ? I am not a windows user, but know there are ports of llvm/clang available, which seems to be a good choice for C projects. Not sure if there are usable ports of gcc.
The reason I am asking is that I am personally not a fan of changing code because of issues that a (proprietary) compiler has. I'd like to focus on gcc and clang mostly because I can easily test with those.

Reply:
MSVCs position on VLAs is "Variable length array (VLA) support isn't planned. VLAs provide attack vectors comparable to gets, which is deprecated and planned for removal"  per [https://learn.microsoft.com/en-us/cpp/build/reference/std-specify-language-standard-version](https://learn.microsoft.com/en-us/cpp/build/reference/std-specify-language-standard-version?view=msvc-170#:~:text=Variable%20length%20array%20(VLA)%20support%20isn%27t%20planned.%20VLAs%20provide%20attack%20vectors%20comparable%20to%20gets%2C%20which%20is%20deprecated%20and%20planned%20for%20removal.) .

There is no doubt the code is not as clean.   Microsofts recommendation would be to use [_malloca](https://learn.microsoft.com/en-us/cpp/c-runtime-library/reference/malloca?view=msvc-170) .  I am happy to add such code (and can gate the code to be MSVC specific still using VLAs for any other compiler), but still will ugly things up a bit.  

> IMO there is no technical difference between VLA and alloca(), the latter feels just a bit clumsy. Or does alloca() check for stack boundaries ?

_malloca does have have the benefit of object alignment and some additional checks.  In addition it falls back onto the heap if space requires it, although it does result in me having to add a _freea at the end of each function (which noops if it wasn't heap allocated).   

> @mitchcapper Is there a technical reason to use MSVC ?

While I have spent my fair share of time with gdb/lldb, the VS debugging experience I have always found to be the most pleasant.  For some of the issues I ran into I am not sure I would have had the patience to track them using LLDB.  Granted I am sure for many it is the exact opposite and completely understand the desire to not support MSVC.  It was fairly clear the choice here was to go in that direction.  There are perfectly good versions of those compilers on windows, and WSL is something many people use heavily without issue.     Here the code wasn't that far from being MSVC friendly again so I figured I would see what the temperature was on bringing support back.   If not, I will just maintain a separate patch for the WIN64LinuxBuild as long as relatively easy to do so.

Reply:
And please check the return value of malloca (might be NULL) :).